signer/src/transaction_signer.rs

```diff
@@ -385,6 +423,31 @@ where
 
         Ok(())
     }
+
+    #[tracing::instrument(skip(self, msg))]
+    async fn send_message(
+        &mut self,
+        msg: impl Into<message::Payload>,
+        bitcoin_chain_tip: &model::BitcoinBlockHash,
+    ) -> Result<(), error::Error> {
+        let msg = msg
+            .into()
+            .to_message(
+                bitcoin::BlockHash::from_slice(bitcoin_chain_tip)
+                    .map_err(error::Error::SliceConversion)?,
+            )
+            .sign_ecdsa(&self.signer_private_key)?;
+
+        self.network.broadcast(msg).await?;
+
+        Ok(())
+    }
+
+    fn signer_pub_key(&self) -> Result<model::PubKey, error::Error> {
+        Ok(p256k1::ecdsa::PublicKey::new(&self.signer_private_key)?
+            .to_bytes()
+            .to_vec())
+    }
```

Comment:
Same for this interface, conceptually this should never fail. No need to change anything here though.

Reply:
Yeah no this should also be infallible in practice.